cmd/jaeger/internal/integration/trace_writer_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap/zaptest"
+)
+
+type MockExporter struct {
+	mock.Mock
+	chunks []ptrace.Traces
+}
+
+func (m *MockExporter) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	m.chunks = append(m.chunks, td)
+	args := m.Called(ctx, td)
+	return args.Error(0)
+}
+
+func (m *MockExporter) Capabilities() consumer.Capabilities {
+	args := m.Called()
+	return args.Get(0).(consumer.Capabilities)
+}
+
+func (*MockExporter) Shutdown(context.Context) error {
+	return nil
+}
+
+func (*MockExporter) Start(context.Context, component.Host) error {
+	return nil
+}
+
+func TestWriteTracesInChunks(t *testing.T) {
+	td := ptrace.NewTraces()
+	resources := td.ResourceSpans()
+
+	// create resources data for td
+	type testSpan struct {
+		name   string
+		spanID [8]byte
+	}
+
+	services := []struct {
```

Comment:
there is no need for this structure - if you do a loop for `i := 0..2` you can generate all the constants via Sprintf